Mathlib/SetTheory/Ordinal/ENat.lean

```diff
@@ -0,0 +1,267 @@
+/-
+Copyright (c) 2024 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios, Yury Kudryashov
+-/
+import Mathlib.SetTheory.Cardinal.ENat
+import Mathlib.SetTheory.Ordinal.Arithmetic
+
+/-!
+# Conversion between `Ordinal` and `ℕ∞`
+
+In this file we define a coercion `Ordinal.ofENat : ℕ∞ → Ordinal`
+and a projection `Ordinal.toENat : Ordinal →*₀o ℕ∞`.
+We also prove basic theorems about these definitions.
+
+## Implementation notes
+
+We define `Ordinal.ofENat` as a function instead of a bundled homomorphism
+so that we can use it as a coercion and delaborate its application to `↑n`.
+
+We define `Ordinal.toENat` as a bundled homomorphism
+so that we can use all the theorems about homomorphisms without specializing them to this function.
+Since it is not registered as a coercion, the argument about delaboration does not apply.
+
+This file is very similar to SetTheory/Cardinal/ENat. Please keep them in sync.
+
+## Keywords
+
+set theory, ordinals, extended natural numbers
+-/
+
+
+open Function Set
+universe u v
+
+namespace Ordinal
+
+/-- Coercion `ℕ∞ → Ordinal`. It sends natural numbers to natural numbers and `⊤` to `ω`.
+
+See also `Ordinal.ofENatHom` for a bundled homomorphism version. -/
+@[coe] def ofENat : ℕ∞ → Ordinal
+  | (n : ℕ) => n
+  | ⊤ => ω
+
+instance : Coe ENat Ordinal := ⟨Ordinal.ofENat⟩
+
+@[simp, norm_cast] lemma ofENat_top : ofENat ⊤ = ω := rfl
+@[simp, norm_cast] lemma ofENat_nat (n : ℕ) : ofENat n = n := rfl
+@[simp, norm_cast] lemma ofENat_zero : ofENat 0 = 0 := rfl
+@[simp, norm_cast] lemma ofENat_one : ofENat 1 = 1 := Nat.cast_one
+
+@[simp, norm_cast] lemma ofENat_ofNat (n : ℕ) [n.AtLeastTwo] :
+    ((no_index (OfNat.ofNat n : ℕ∞)) : Ordinal) = OfNat.ofNat n :=
```

Comment:
Per [note](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Nat/Cast/Defs.html#Mathlib.LibraryNote.%C2%ABno_index%20around%20OfNat.ofNat%C2%BB), it seems like `ofNat(n)` is preferred?